modules/apps/transfer/keeper/keeper.go

```diff
@@ -328,3 +328,16 @@ func (k Keeper) GetForwardedPacket(ctx sdk.Context, portID, channelID string, se
 
 	return storedPacket, true
 }
+
+// deleteForwardedPacket deletes the forwarded packet from the store.
+func (k Keeper) deleteForwardedPacket(ctx sdk.Context, portID, channelID string, sequence uint64) error {
+	store := ctx.KVStore(k.storeKey)
+	packetKey := types.PacketForwardKey(portID, channelID, sequence)
+	if !store.Has(packetKey) {
+		return fmt.Errorf("packet with PortID %s, channelID %s and sequence %d not found in the store", portID, channelID, sequence)
+	}
```

Comment:
I think logically this shouldn't be possible given that we only end up in `acknowledgeForwardedPacket` if the packetKey existed. Typically delete functions in our codebase don't return an error as the function call is never expected to fail. I have a very minor preference for not returning an error (skipping the has check), but I'm very happy to go with either decision. 

If we decide to return an error, it should use a registered error type otherwise the SDK will discard the error message with an internal error when returning to the relay message (I believe in this [else](https://github.com/cosmos/cosmos-sdk/blob/main/errors/abci.go#L65) statement)

Reply:
> If we decide to return an error, it should use a registered error type otherwise the SDK will discard the error message with an internal error when returning to the relay message (I believe in this [else](https://github.com/cosmos/cosmos-sdk/blob/main/errors/abci.go#L65) statement)

Ah, thanks for the pointer!


I don't feel strongly on the error. I see that `store.Delete()` doesn't return any error and just panics if there is no valid key. I'm happy to skip the `has` check.